x/stake/cross_stake/cross_stake.go

```diff
@@ -0,0 +1,340 @@
+package cross_stake
+
+import (
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/bsc"
+	"github.com/cosmos/cosmos-sdk/bsc/rlp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+)
+
+type CrossStakeApp struct {
+	stakeKeeper Keeper
+}
+
+func NewCrossStakeApp(stakeKeeper Keeper) *CrossStakeApp {
+	return &CrossStakeApp{
+		stakeKeeper: stakeKeeper,
+	}
+}
+
+func (app *CrossStakeApp) ExecuteAckPackage(ctx sdk.Context, payload []byte) sdk.ExecuteResult {
+	if len(payload) == 0 {
+		app.stakeKeeper.Logger(ctx).Info("receive cross stake ack package")
+		return sdk.ExecuteResult{}
+	}
+
+	eventCode, pack, err := DeserializeCrossStakeSynPackage(payload)
+	if err != nil {
+		app.stakeKeeper.Logger(ctx).Error("unmarshal cross stake ack package error", "err", err.Error(), "package", string(payload))
+		panic("unmarshal cross stake ack package error")
+	}
+
+	var result sdk.ExecuteResult
+	switch eventCode {
+	case types.CrossStakeTypeDistributeReward:
+		result, err = app.handleDistributeRewardRefund(ctx, pack.(types.CrossStakeDistributeRewardSynPackage))
+	case types.CrossStakeTypeDistributeUndelegated:
+		result, err = app.handleDistributeUndelegatedRefund(ctx, pack.(types.CrossStakeDistributeUndelegatedSynPackage))
```

Comment:
We need handle the funds that is stuck in Beacon chain.

Reply:
Please throw error if the package type is unknown.

Reply:
Done